src/groups/group_action.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    AbstractActionOnManifold
+
+An abstract group action on a manifold.
+"""
+abstract type AbstractActionOnManifold end
+
+"""
+    base_group(A::AbstractActionOnManifold)
+
+The group that acts in action `A`.
+"""
+function base_group(A::AbstractActionOnManifold)
+    error("Function base_group is not yet defined for type $(typeof(A)).")
+end
+
+"""
+    action_on(A::AbstractActionOnManifold)
+
+The manifold the action `A` acts upon.
+"""
+function action_on(A::AbstractActionOnManifold)
+    error("Function action_on is not yet defined for type $(typeof(A)).")
+end
+
+"""
+    apply_action(A::ActionOnManifold, x, a)
+
+Apply action `a` of type `A` to the point `x`.
```

Comment:
This indicates that `a::A`, but I don't think that's what you mean?

Reply:
Right, it's a misleading description.